src/main/java/com/alex/eshop/keycloak/KeycloakService.java

```diff
@@ -25,4 +40,63 @@ public AccessTokenResponse getToken() {
         AuthzClient authzClient = AuthzClient.create(authzConf);
         return authzClient.obtainAccessToken();
     }
+
+    public UserRepresentation getUserRepresentation(String userUuid) {
+        UserRepresentation userRepresentation = keycloakClientFactory.getInstance()
+                .realm(applicationProperties.getKeycloak().getRealm())
+                .users()
+                .get(userUuid)
+                .toRepresentation();
+        return userRepresentation;
+    }
+
+    public UserRepresentation createUser(UserRegisterDTO userRegisterDTO) {
+        UsersResource usersResource = keycloakClientFactory.getInstance()
+                .realm(applicationProperties.getKeycloak().getRealm())
+                .users();
+
+        List<RoleRepresentation> realmRoles = new ArrayList<>();
+
+        realmRoles.add(keycloakClientFactory.getInstance()
+                .realm(applicationProperties.getKeycloak().getRealm())
+                .roles().get("ROLE_USER").toRepresentation());
+
+        if (!Objects.equals(userRegisterDTO.getPassword(), userRegisterDTO.getConfirmPassword())) {
+            throw new InvalidData("Invalid password");
+        }
+
+        CredentialRepresentation credentialRepresentation = createPasswordCredentials(userRegisterDTO.getPassword());
+
+        UserRepresentation userRepresentation = new UserRepresentation();
+
+        ZonedDateTime zonedDateTime = ZonedDateTime.of(LocalDateTime.now(), ZoneId.systemDefault());
+        Long createdTimestamp = zonedDateTime.toInstant().toEpochMilli();
+
+        userRepresentation.setUsername(userRegisterDTO.getUserName());
+        userRepresentation.setEmail(userRegisterDTO.getEmail());
+        userRepresentation.setFirstName(userRegisterDTO.getFirstName());
+        userRepresentation.setLastName(userRegisterDTO.getLastName());
+        userRepresentation.setCredentials(Collections.singletonList(credentialRepresentation));
+        userRepresentation.setEnabled(true);
+        userRepresentation.setCreatedTimestamp(createdTimestamp);
+
+        if (usersResource.create(userRepresentation).getStatus() != 201) {
+            throw new Conflict("Problems occurred while creating user");
+        }
+
+        String userUuid = keycloakClientFactory.getInstance()
+                .realm(applicationProperties.getKeycloak().getRealm())
+                .users()
+                .search(userRegisterDTO.getUserName()).get(0).getId();
```

Comment:
in `search()` function second parameter "exact" must be "true", because it cant find other user with similar name